### PR TITLE
[release/6.5] Build and test using .NET 6

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

**Comment by CZEMacLeod** [comment]
I'm assuming that should be release/6.5 not realease/6.5?

**Comment by bricelam** [comment]
Man, Microsoft isn't playing around with this Zero Trust initiative.
 
image

File: src/EntityFramework.NuGet/EntityFramework.NuGet.csproj [hunk 1]
```diff
@@ -2,7 +2,8 @@
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
 
   <PropertyGroup>
-    <TargetFrameworks>net45;net40;$(DefaultNetCoreTargetFramework)</TargetFrameworks>
+    <TargetFrameworks>net45;net40;netcoreapp3.0</TargetFrameworks>
```
Comment (wtgodbe) [inline src/EntityFramework.NuGet/EntityFramework.NuGet.csproj:5]: Why netcoreapp3.0?

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -1,11 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Dependencies>
-  <ProductDependencies>
-    <Dependency Name="Microsoft.NETCore.App.Internal" Version="3.1.0-rtm.19565.2" Pinned="true">
-      <Uri>https://github.com/dotnet/core-setup</Uri>
-      <Sha>65f04fb6db7a5e198d05dbebd5c4ad21eb018f89</Sha>
-    </Dependency>
-  </ProductDependencies>
   <ToolsetDependencies>
     <Dependency Name="Microsoft.DotNet.Arcade.Sdk" Version="8.0.0-beta.23110.3">
```
Comment (wtgodbe) [inline eng/Version.Details.xml:4]: This should be changed to match the entry in `global.json`. I recommend adding a `darc` subscription from Arcade's 6.0 branch, then manually triggering it to see what it updates. Ping me offline if you need help with that

File: src/EntityFramework.NuGet/EntityFramework.NuGet.csproj (hunk 1 above)
Comment (bricelam) [reply src/EntityFramework.NuGet/EntityFramework.NuGet.csproj:5]: Well, technically, it should be netstandard2.1, but then it couldn't depend on ef6.csproj (which is also netcoreapp3.0)

The actual package's targets are specified in a custom .nuspec file.

File: eng/Version.Details.xml (hunk 2 above)
Comment (bricelam) [reply eng/Version.Details.xml:4]: Thanks, yeah I want to talk to you about which channels this and main should participate in.

Comment (bricelam) [reply eng/Version.Details.xml]: Updated. Here are the current subscriptions.

Branch | Channel | Description
--- | --- | ---
main | .NET Eng - Latest | A hypothetical future version
release/6.5 | .NET 6 Eng | The version we're trying to ship
release/6.4 | .NET 3 Eng | Servicing branch for 6.4 (shipped with .NET 3.1)

**Review by ajcvickers — APPROVED** [review]

**Comment by bricelam** [comment]
> Man, Microsoft isn't playing around with this Zero Trust initiative.
> 
> image

@ajcvickers @danmoseley Can I get permissions merge to the `release/6.5` branch?

**Review by wtgodbe — APPROVED** [review]

**Comment by bricelam** [comment]
@wtgodbe Do you have admin privileges? Can you check the rules on the `release/6.5` branch?

**Comment by wtgodbe** [comment]
I do - I gave you perms to merge into `release/*` branches, let me know if that didn't work

**Pull request merged**